diff --git a/node_modules/@graywolfai/esbuild-register/dist/node.js b/node_modules/@graywolfai/esbuild-register/dist/node.js
index 765f4d7..3984a1a 100644
--- a/node_modules/@graywolfai/esbuild-register/dist/node.js
+++ b/node_modules/@graywolfai/esbuild-register/dist/node.js
@@ -10,6 +10,8 @@ var atob2 = require('atob');
 var fs2 = require('fs');
 var JoyCon = require('joycon');
 var strip = require('strip-json-comments');
+var util = require("util")
+var TextDecoder = util.TextDecoder;
 
 function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }
 
