diff --git a/node_modules/vite/dist/node/resolver.js b/node_modules/vite/dist/node/resolver.js
index 0688f5b..f51077e 100644
--- a/node_modules/vite/dist/node/resolver.js
+++ b/node_modules/vite/dist/node/resolver.js
@@ -112,7 +112,7 @@ function createResolver(root, resolvers = [], userAlias = {}) {
                         }
                     },
                     fileToRequest(filePath) {
-                        if (filePath.startsWith(target + pathSeparator)) {
+                        if (filePath.startsWith(target + pathSeparator) && !filePath.endsWith("/@react-refresh")) {
                             return slash_1.default(key + path_1.default.relative(target, filePath));
                         }
                     }
