diff --git a/node_modules/vite/dist/node/build/buildPluginAsset.js b/node_modules/vite/dist/node/build/buildPluginAsset.js
index b8b434f..ff5bbf4 100644
--- a/node_modules/vite/dist/node/build/buildPluginAsset.js
+++ b/node_modules/vite/dist/node/build/buildPluginAsset.js
@@ -60,6 +60,7 @@ exports.resolveAsset = async (id, root, publicBase, assetsDir, inlineLimit) => {
 exports.registerAssets = (assets, bundle) => {
     for (const [fileName, source] of assets) {
         bundle[fileName] = {
+            name: fileName,
             isAsset: true,
             type: 'asset',
             fileName,
diff --git a/node_modules/vite/dist/node/build/buildPluginCss.js b/node_modules/vite/dist/node/build/buildPluginCss.js
index 2aa0331..22fff0b 100644
--- a/node_modules/vite/dist/node/build/buildPluginCss.js
+++ b/node_modules/vite/dist/node/build/buildPluginCss.js
@@ -124,6 +124,7 @@ exports.createBuildCssPlugin = ({ root, publicBase, assetsDir, minify = false, i
             }
             const cssFileName = `style.${hash_sum_1.default(css)}.css`;
             bundle[cssFileName] = {
+                name: cssFileName,
                 isAsset: true,
                 type: 'asset',
                 fileName: cssFileName,
diff --git a/node_modules/vite/dist/node/esbuildService.js b/node_modules/vite/dist/node/esbuildService.js
index d4bc426..b487ca7 100644
--- a/node_modules/vite/dist/node/esbuildService.js
+++ b/node_modules/vite/dist/node/esbuildService.js
@@ -53,7 +53,7 @@ exports.transform = async (src, request, options = {}, jsxOption) => {
         loader: options.loader || path_1.default.extname(file).slice(1),
         sourcemap: true,
         sourcefile: request,
-        target: 'es2019'
+        target: 'es2020'
     };
     try {
         const result = await service.transform(src, options);
