diff --git a/node_modules/mobicon/index.js b/node_modules/mobicon/index.js
index 214f0ef..10650df 100644
--- a/node_modules/mobicon/index.js
+++ b/node_modules/mobicon/index.js
@@ -80,12 +80,15 @@ module.exports = (file, opts) => {
 
 			return Promise.all(icons.map(icon => {
 				const dest = path.join(opts.dest, icon.file);
-				const dimension = calculateDimension(size, icon.dimension, opts, resizeFn);
+				const dimension = calculateDimension(size, icon.dimension, { ...opts, contentRatio: 1 }, resizeFn);
+				const dimension2 = calculateDimension(size, icon.dimension, opts, resizeFn);
 
-				const image = gm(file)[resizeFn](dimension.width, dimension.height)
+				const image = gm(file)
 					.gravity('Center')
 					.background(opts.background)
-					.extent(icon.dimension, icon.dimension);
+					[resizeFn](dimension.width, dimension.height)
+					.extent(icon.dimension, icon.dimension)
+					
 
 				return mkdirp(path.dirname(dest))
 					.then(() => pify(image.write.bind(image))(dest))
@@ -97,7 +100,9 @@ module.exports = (file, opts) => {
 									return execa('gm', ['composite', '-compose', 'in', dest, maskLocation, dest]);
 								});
 						}
-					});
+					})
+					.then(() => gm(dest)[resizeFn](dimension2.width, dimension2.height).gravity('Center').background("transparent").extent(icon.dimension, icon.dimension))
+					.then((image) => pify(image.write.bind(image))(dest));
 			}));
 		});
 };
