diff --git a/node_modules/esbuild-register/dist/node.js b/node_modules/esbuild-register/dist/node.js
index 777ed98..295abf6 100644
--- a/node_modules/esbuild-register/dist/node.js
+++ b/node_modules/esbuild-register/dist/node.js
@@ -52,12 +52,13 @@ function installSourceMapSupport() {
 const DEFAULT_EXTENSIONS = ['.js', '.jsx', '.ts', '.tsx', '.mjs'];
 function compile(code, filename) {
     const options = getOptions(path.dirname(filename));
+    const name = path.basename(filename);
     const { js, warnings, jsSourceMap } = esbuild.transformSync(code, {
         sourcefile: filename,
         sourcemap: true,
         // Just use tsx for everything until we find a way to let user configure this
-        loader: 'tsx',
-        target: options.target,
+        loader: name.substr(name.lastIndexOf(".") + 1),
+        target: "es2017",
         jsxFactory: options.jsxFactory,
         jsxFragment: options.jsxFragment,
     });
